src/main/scala/cognite/spark/v1/SequencesRelation.scala

```diff
@@ -155,19 +164,26 @@ class SequencesRelation(config: RelationConfig)(val sqlContext: SQLContext)
   }
 
   override def upsert(rows: Seq[Row]): IO[Unit] = {
-    val sequences =
-      rows
-        .map(fromRow[SequenceUpsertSchema](_))
+    val sequences = rows.map(fromRow[SequenceUpsertSchema](_))
 
     implicit val toCreate = transformerUpsertToCreate()
 
     implicit val toUpdate = transformerUpsertToUpdate(sequences)
 
-    genericUpsert[Sequence, SequenceUpsertSchema, SequenceCreate, SequenceUpdate, SequencesResource[IO]](
+    val groupedSequences: Seq[Vector[SequenceUpsertSchema]] = splitSequences[SequenceUpsertSchema](
       sequences,
-      isUpdateEmpty,
-      client.sequences
-    )
+      (s: SequenceUpsertSchema) => s.columns.map(_.size).getOrElse(0))
+
+    // scalastyle:off no.whitespace.after.left.bracket
+    groupedSequences.toList.map { sequencesToCreate =>
+      genericUpsert[
+        Sequence,
+        SequenceUpsertSchema,
+        SequenceCreate,
+        SequenceUpdate,
+        SequencesResource[IO]](sequencesToCreate, isUpdateEmpty, client.sequences)
+    }.sequence_
```

Comment:
traverse? :D :D

Reply:
traverse_ :p

Reply:
[It's always Traverse](https://github.com/cognitedata/cdp-spark-datasource/pull/612/commits/f875f3a8f5e3b411a7db816f16bac66cc430f1be)